### ISSUE TITLE
support do catch

### ISSUE THREAD

**Issue by elbaro** [intellij-rust/intellij-rust#2400]
```rust
#![feature(catch_expr)]

let a = do catch {
    let b = f()?;
    b.c()
};
```
The code within `do catch` scope has broken highlight and intellisens.

**Comment by mehcode** [comment]
https://github.com/rust-lang/rfcs/pull/2388 was merged so the expression is now

```rust
let a = try {
  let b = f()?;

  // ok wrapping has not been decided yet so if the RFC was 
  // implemented as-is values would not be ok wrapped 
  Ok(b.c())
}
```

**Comment by Centril** [comment]
Do note that `try { .. }` currently Ok-wraps in nightly (and the RFC specifies this).

**Comment by breeswish** [comment]
+1 for this feature.

**Comment by scottmcm** [comment]
As of https://github.com/rust-lang/rust/pull/52602, `try` is now a keyword in 2018 rust (though the actual `try{}` expression is still unstable, requiring nightly and the appropriate feature gate).
